### PR TITLE
Release tracking PR: `secp256k1-sys 0.10.0`

### PR DESCRIPTION
**Pull request by tcharding**
Vendor latest tagged version of `secp256k1` and prepare for release:

- Bump the version number to `0.10.0`
- Run the vendor script (vendoring `secp256k1 0.4.1`)
- Update lock files
- Add changelog entry
- Depend on new version in `secp256k1/Cargo.toml`

### CODE REVIEW

**Comment by tcharding** [comment]
Nothing is ever as simple as it seems, if we do this we will have to do another release of secp-sys as well.

**Comment by apoelstra** [comment]
Heh, we might as well leave secp-sys at 1.48 then. It's okay if it's behind the other crates.

**Comment by tcharding** [comment]
Ok, I'll spilt out another CI job for 1.48 just for secp-sys

**Comment by tcharding** [comment]
Bother, `cargo 1.48` tries to parse the top level manifest still so chokes on the `edition = 2021` even though we are only testing `secp256k1-sys`.

**Comment by tcharding** [comment]
Just in case you lost context for this PR with the name changes, the problem with leaving `secp256k1-sys` on 1.48.0 was that `cargo test` didn't run in the `secp256k1-sys/` directory because `cargo` attempts to parse to top level manifest.

File: secp256k1-sys/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# 0.10.0 - 2024-03-27
+
+* Bump MSRV to Rust `v0.51.1` [#688](https://github.com/rust-bitcoin/rust-secp256k1/pull/688)
```
Comment (sanket1729) [inline secp256k1-sys/CHANGELOG.md:3]: Change version here.

**Review by sanket1729 — COMMENTED** [review]
One comment. Rest looks good

**Comment by apoelstra** [comment]
> Nothing is ever as simple as it seems, if we do this we will have to do another release of secp-sys as well.

Actually I'm a bit confused -- what is "this" here? If you mean, revendoring libsecp requires us to release a new version of secp-sys, then yes, this is true. And we ought to do that before releasing a new version of rust-secp. And if we're doing a release, why wouldn't we bump the MSRV at the same time?

I had initially read your comment as "if we bump the MSRV then we need to release a new secp-sys version, which we could avoid if we didn't bump the MSRV". But it seems like we need a new version regardless, because of the revendoring (which we should definitely do, there have been some further sidechannel defenses added upstream).

**Comment by apoelstra** [comment]
Oh, I understand, we're revendoring *only* to change the version number.

Why don't we just update to the latest version of libsecp256k1?

**Comment by tcharding** [comment]
> Why don't we just update to the latest version of libsecp256k1?

I wondered that myself, was just artificially limiting scope, and I was lazy, I see now that lates is 0.4.1 - we should definitely upgrade to that. Will have another go.

**Comment by apoelstra** [comment]
Sounds good -- I think we should update to 0.4.1, do a new secp-sys release, bump the MSRV of secp-sys while we're at it, and then we have a release that is clearly worthwhile and no questions about whether we should try to hack around the need for it.

File: secp256k1-sys/CHANGELOG.md (hunk 1 above)
Comment (tcharding) [reply secp256k1-sys/CHANGELOG.md:3]: OMG what the hell is that! Thanks man.

**Comment by tcharding** [comment]
MSRV change is over in #693 now, I'll do the vendoring here still because its tied into the version number. And use 0.4.1

**Comment by tcharding** [comment]
Doing #693 separately may have been an overkill but now I've used it in the CHANGLOG, it serves some purpose (maybe) as a historical record.

**Comment by apoelstra** [comment]
When I run the script locally I find four files that you committed that aren't committed on my end:
```
        supprimé :        depend/secp256k1/TAGS
        supprimé :        depend/secp256k1/src/libsecp256k1-config.h
        supprimé :        depend/secp256k1/src/libsecp256k1-config.h.in
        supprimé :        depend/secp256k1/src/stamp-h1
```
(Supprimé means "deleted" in French.)

This happens both directly on my system and in my local CI.

**Comment by tcharding** [comment]
No worries, I'm not going to get to this today but I will at some stage.

You speak French as well as English with no accent eh, that explains your grammar skills.

**Comment by apoelstra** [comment]
I can *read* French, though I am missing a lot of vocabulary and struggle to understand even simple news articles. I keep my locale set to `fr_CA` mostly to fuzz my software and make sure it doesn't do something crazy/broken with a non-English local.

But I cannot speak or write French. :)

**Comment by tcharding** [comment]
> I keep my locale set to fr_CA mostly to fuzz my software and make sure it doesn't do something crazy/broken with a non-English local.

Lol, things like this remind me you are a good bloke to have on the team.

**Comment by tcharding** [comment]
Is there a reason we haven't merged this PR and release `secp256k1-sys v.010.0` - is this waiting on anything from me?

**Comment by apoelstra** [comment]
>Is there a reason we haven't merged this PR and release secp256k1-sys v.010.0 - is this waiting on anything from me?

I was waiting for it to show up in my github notifications again, indicating that everything this was blocked on had merged, and that you'd noticed so I didn't have to :).

**Comment by apoelstra** [comment]
> When I run the script locally I find four files that you committed that aren't committed on my end:
> 
> ```
>         supprimé :        depend/secp256k1/TAGS
>         supprimé :        depend/secp256k1/src/libsecp256k1-config.h
>         supprimé :        depend/secp256k1/src/libsecp256k1-config.h.in
>         supprimé :        depend/secp256k1/src/stamp-h1
> ```
> 
> (Supprimé means "deleted" in French.)
> 
> This happens both directly on my system and in my local CI.

All of these files are still present.

**Comment by tcharding** [comment]
Jee wiz, spent so much time talking about your French skills I forgot to actually do the fix ... its hard to get good help.

**Review by apoelstra — APPROVED** [review]
ACK 2bba8f9f5a7ea8af619a0df0afb4f36f3e00013a

**Comment by apoelstra** [comment]
Tagged and published.

**Pull request merged**